sds_data_manager/lambda_code/IAlirtCode/ialirt_ingest.py

```diff
@@ -421,6 +428,122 @@ def insert_data(data: list[dict], algorithm_table, instrument: str):
+        Reformatted science data.
+    hk_data : list[dict]
+        Reformatted housekeeping data.
+    """
+    # Reformat data (remove all keep/exclude keys except hk
+    # once imap_processing is updated)
+    exclude_keys = {"apid", "met", "mag_hk_status"}
+    rename_map = {"met_in_utc": "time_utc"}
+    science_data = [
+        {rename_map.get(k, k): v for k, v in item.items() if k not in exclude_keys}
+        for item in data
+    ]
+    keep_keys = {"met_in_utc", "instrument", "mag_hk_status"}
+    hk_data = [
+        {
+            rename_map.get(k, k): ("mag_hk" if k == "instrument" and v == "mag" else v)
+            for k, v in item.items()
+            if k in keep_keys
+        }
+        for item in data
+        if item.get("instrument") == "mag"
+    ]
+
+    return science_data, hk_data
+
+
+def insert_formatted_data(
+    data: list[dict],
+    data_table,
+    instrument: str,
+):
+    """Insert database rows.
+
+    Parameters
+    ----------
+    data : list[dict]
+        Data product produced from processing respectively instrument.
+    data_table : dynamodb.Table
+        The DynamoDB table to insert or update the data.
+    instrument : str
+        The prefix for the product name.
+    """
+    # Get time range.
+    times = [item["met_in_utc"] for item in data]
+    min_time = min(times)
+    max_time = max(times)
+    logger.info(f"Processing {min_time} to {max_time}.")
+
+    science_data, hk_data = reformat_data(data)
+
+    # Query existing items.
+    response = data_table.query(
+        KeyConditionExpression=Key("instrument").eq(instrument)
+        & Key("time_utc").between(min_time, max_time)
+    )
+    hk_response = data_table.query(
+        KeyConditionExpression=Key("instrument").eq(f"{instrument}_hk")
+        & Key("time_utc").between(min_time, max_time)
+    )
+
+    existing_items = {item["time_utc"]: item for item in response.get("Items", [])}
+    existing_hk_items = {
+        item["time_utc"]: item for item in hk_response.get("Items", [])
+    }
+
+    # Insert science data
+    for raw in science_data:
+        time = raw["time_utc"]
+        existing = existing_items.get(time)
+
+        if not existing:
+            data_table.put_item(Item=raw)
+        logger.info(f"Inserted {instrument.upper()}.")
+
+    # Insert hk data
+    if hk_data:
+        for raw in hk_data:
+            time = raw["time_utc"]
+            existing = existing_hk_items.get(time)
+
+            if not existing:
+                data_table.put_item(Item=raw)
+            logger.info(f"Inserted Housekeeping for {instrument.upper()}.")
+
+    # Calculate the spacecraft position and velocity in GSE/GSM coordinates.
+    et = str_to_et(min_time)
+    gsm_state = imap_state(
+        [et], ref_frame=SpiceFrame.IMAP_GSM, observer=SpiceBody.EARTH
+    )
+    gse_state = imap_state(
+        [et], ref_frame=SpiceFrame.IMAP_GSE, observer=SpiceBody.EARTH
+    )
+    geolocation = {
+        "instrument": "geolocation",
```

Comment:
what about calling this "spacecraft", "sc_pos_vel" or something similar? geolocation is usually something on Earth, but we are outside of Earth's orbit.